packages/api/cms-api/src/kubernetes/kubernetes.service.ts

```diff
@@ -47,67 +47,32 @@ export class KubernetesService {
     }
 
     async getCronJob(name: string): Promise<V1CronJob> {
-        const { response, body } = await this.batchApi.readNamespacedCronJob(name, this.namespace);
-        if (response.statusCode !== 200) {
-            throw new Error(`Error fetching Job "${name}"`);
-        }
-        return body;
+        return this.batchApi.readNamespacedCronJob({ name, namespace: this.namespace });
     }
 
     async getAllCronJobs(labelSelector: string): Promise<V1CronJob[]> {
-        const { response, body } = await this.batchApi.listNamespacedCronJob(
-            this.namespace,
-            undefined,
-            undefined,
-            undefined,
-            undefined,
-            labelSelector,
-        );
-        if (response.statusCode !== 200) {
-            throw new Error(`Error fetching CronJobs for selector "${labelSelector}"`);
-        }
-
-        return body.items;
+        const { items } = await this.batchApi.listNamespacedCronJob({ namespace: this.namespace, labelSelector });
+        return items;
     }
 
     async deleteJob(name: string): Promise<void> {
-        const { response } = await this.batchApi.deleteNamespacedJob(name, this.namespace, undefined, undefined, undefined, undefined, "Background");
-        if (response.statusCode !== 200) {
+        const { status } = await this.batchApi.deleteNamespacedJob({ name, namespace: this.namespace, propagationPolicy: "Background" });
+        if (status !== undefined && status !== "Success") {
```

Comment:
`status` in theory can be `undefined`, which would be an unknown status. Should we throw an error if the status is undefined?

Reply:
It now throws an error: 149c18fc68c0ac0fd17470515494a5c14d74bdb5